include/deal.II/lac/trilinos_tpetra_vector.h

```diff
@@ -759,15 +853,40 @@ namespace LinearAlgebra
       create_tpetra_comm_pattern(const IndexSet &source_index_set,
                                  const MPI_Comm  mpi_comm);
 
+      /**
+       * The vector is compressed if and only if the underlying index set is a
+       * one-to-one map.
+       *
+       * If the vector is compressed (i.e., compressed == true), it can only
+       * access and modify entries included in the locally owned index set.
+       * And if the vector is not compressed (i.e., compressed == false), it
+       * can access and modify entries included in the locally relevant index
+       * set.
+       */
+      bool compressed;
+
       /**
        * Teuchos::RCP to the actual Tpetra vector object.
        */
       Teuchos::RCP<VectorType> vector;
 
+      /**
+       * Pointer to the Tpetra::Map storing the owned elemenets
+       * per rank. This map has a one-to-one mapping.
+       */
+      Teuchos::RCP<MapType> owned_elements;
+
+      /**
+       * Pointer to the Tpetra::Map storing the relevant elements
+       * per rank. There are entries in this map, that belong to multiple
+       * ranks.
+       */
+      Teuchos::RCP<MapType> relevant_elements;
```

Comment:
Would it be possible to call these variables `locally_(owned|relevant)_elements` to stick with the usual terminology?

Reply:
Sure!
But, I have to rename the function `locally_owned_elements(`) to `get_locall_owned_elements()`.

Reply:
Yes, I like this name better.

Reply:
Renamed!